Rectangle/AppDelegate.swift

```diff
@@ -555,17 +555,36 @@ extension AppDelegate {
             func getUrlName(_ name: String) -> String {
                 return name.map { $0.isUppercase ? "-" + $0.lowercased() : String($0) }.joined()
             }
+            func getAppBundleId(_ components: URLComponents) -> String? {
+                let appBundleIdParam = (components.queryItems?.first { $0.name == "app-bundle-id" })
+                // if param not present the behavior is to use front app
+                return if let appBundleIdParam {
+                    appBundleIdParam.value
+                } else {
+                    ApplicationToggle.frontAppId
+                }
+            }
             for url in urls {
-                guard 
+                guard
                     let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
-                    components.host == "execute-action",
-                    components.path.isEmpty,
-                    let name = (components.queryItems?.first { $0.name == "name" })?.value,
-                    let action = (WindowAction.active.first { getUrlName($0.name) == name })
+                    components.path.isEmpty
                 else {
                     continue
                 }
-                action.postUrl()
+                    
+                let name = (components.queryItems?.first { $0.name == "name" })?.value
+                let appBundleId = getAppBundleId(components)
```

Comment:
Here's a totally subjective comment, and I'm good with this as-is, but I actually like calling this inline in the cases below like this:
```
self.applicationToggle.disableApp(appBundleId: getAppBundleId(components))
```
No need to change it if you don't want to.